### DISCUSSION TITLE
Gestion lorsque la commune est sur 2 zones

### DISCUSSION THREAD

**Discussion by OUARZA** [OUARZA/Propluvia#54] [Ideas]
je vis entre 2 zones amont/aval, est-ce que les 2 retours dans les logs sont sauvegardés successivement?

Code INSEE pour test : 24557 (24750 Trélissac)

**Comment by elmerdu69** [comment]
je n'ai pas bien compris cette demande
Il a créé 2 équipements avec 2 communes différentes ? auquel cas dans le log il a bien d'enregistré les infos pour les 2 communes

**Reply by OUARZA** [reply]
Non, il arrive qu'une commune soit sur 2 zones différentes.
![image](https://github.com/OUARZA/Propluvia/assets/34892335/13c65f8a-4ebf-4c8c-ad38-c0eab9ced67b)

**Reply by elmerdu69** [reply]
OK compris
Ce cas n'est pas traité. Il me faut revoir la logique de récupération des données. Et peut être qu'il faudra ajouter un cas d'affichage pour le widget.
Je le traiterai après tout le reste

**Reply by OUARZA** [reply]
Aucun problème :)
En toute logique, le logement doit être sur une seule zone.
Je sais pas si c'est possible mais quand on renseigne le code INSEE et qu'on sauvegarde, avoir un menu déroulant où on choisit la zone concernée.
Je te laisse regarder ça plus tard ;)

**Comment by naos44** [comment]
Bonjour,

Je teste le plugin et effectivement il y a un problème sur le choix des zones. La ville de Nantes est sur 4 zones différentes par contre l'eau courante est prélevée à un seul endroit (Bassin de la Loire).
A chaque refresh la zone semble être choisie aléatoirement.
J'ai contourné le problème en choisissant une commune qui se trouve sur une seule zone du Bassin de la Loire.

Bravo pour le plugin en tout cas  je vais pouvoir m'en servir pour adapter mes durées d'arrosage

**Reply by elmerdu69** [reply]
Bonjour @naos44 
Merci pour ton message.
Le plugin ne gère actuellement pas quand il y a plusieurs zone d'eau de surface dans une même commune. C'est dû au fait que quand on a construit l’algorithme de récupération des données, ce cas d'usage n'a pas été pris en compte car pas connu de notre part.
Je dois donc revoir en profondeur toute cette partie là et peut être mettre en place un mécanisme de sélection.
Je ne vais pas m'y atteler tout de suite car je veux terminer les autres points ouverts. Mais c'est dans la todo.

PS : avec @OUARZA , on préfère avoir les discutions et changes avec les utilisateurs sur community et garder ce canal de discutions ici pour les échanges entre nous sur les différents points ouverts. merci de ta compréhension
